### PR TITLE
ui: Update Locality hierarchy on Node List 

### LINKED ISSUE #44919: ui: Node list locality hierarchy update
Related to #42923

- [ ] Display the lowest tier locality in the node list tree

- [ ] On hover, display a tool tip showing the full locality tree. Designs can be seen here: https://zpl.io/2pX8gej

Screen Shot 2020-02-10 at 11 43 54 AM

Screenshots of the current behavior for reference:
Screen Shot 2020-02-06 at 3 43 16 PM

Screen Shot 2020-02-06 at 3 43 30 PM
Screen Shot 2020-02-10 at 11 43 54 AM

### PR DESCRIPTION
**Pull request by koorosh**
Fixes: https://github.com/cockroachdb/cockroach/issues/44919
Depends on: https://github.com/cockroachdb/cockroach/pull/45007

Before, Node list with geo partitioned nodes was
grouped by 'region' key what was incorrect due to
locality keys might have arbitrary names and it
doesn't represent the lowest locality tier.

Now, nodes are grouped by lowest level tier and
also display all tiers in tooltip.

Preview:
Screenshot 2020-02-12 at 23 13 03

### CODE REVIEW

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 1]
```diff
@@ -133,18 +137,30 @@ export class NodeList extends React.Component<LiveNodeListProps> {
...
+            </Link>
+          );
         } else {
           // Top level grouping item does not have nodeId
           return (
-            <React.Fragment>
-              <Text>{record.region}</Text>
-            </React.Fragment>);
+            <Text>
+              <Tooltip
```
Comment (dhartunian) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: Would be nice to extract the tooltip to a pure presentation component

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 2]
```diff
@@ -375,9 +388,18 @@ export const liveNodesTableDataSelector = createSelector(
...
+        // element in the group because grouping is made by string composed
+        // from location values.
+        const firstNodeInGroup = nodesPerRegion[0];
+        const tiers = getNodeLocalityTiers(firstNodeInGroup);
+        const location = _.last(tiers);
+
         return {
           key: `${regionKey}`,
-          region: regionKey,
+          region: location?.value,
```
Comment (dhartunian) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: why do we call this `location`? Why not `lastTier`?

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 3]
```diff
@@ -133,18 +137,30 @@ export class NodeList extends React.Component<LiveNodeListProps> {
       render: (_text, record) => {
         if (!!record.nodeId) {
           return (
-            <React.Fragment>
-              <Link className="nodes-table__link" to={`/node/${record.nodeId}`}>
-                <Text textType={TextTypes.BodyStrong}>{`N${record.nodeId} `}</Text>
-                <Text>{record.region}</Text>
-              </Link>
-            </React.Fragment>);
+            <Link className="nodes-table__link" to={`/node/${record.nodeId}`}>
```
Comment (dhartunian) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: Can this `<Link>` tag be extracted to match the component on line 303 below?

**Review by dhartunian — CHANGES_REQUESTED** [review]

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx (hunk 1 above)
Comment (koorosh) [reply pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: Sorry if I wrong, but Tooltip component is already defined as pure component.
In this case we provide only props (`title` and `children`) as inputs. Both of these props has
to be defined outside of tooltip component.
To make it more readable, I would extract entire render component.

Comment (koorosh) [reply pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: @dhartunian , please check if new changes somehow fit to yours proposed changes 🙄

**Review by dhartunian — APPROVED** [review]

**Pull request merged**